### PR TITLE
MSEARCH-829: [Linked Data] Enhanced Search & filter

### PR DESCRIPTION
**Pull request by PBobylev**
### Purpose
Introduce new Linked Data Instance index with endpoint. Extend existed Linked Data Work index with more fields.

### Approach
New resource model, index and search endpoint introduction (LD instance). Existed model, index enhancement (LD work).

### Changes Checklist
- [x] **API Changes**: Document any API paths, methods, request or response bodies changed, added, or removed.
- [x] **Unit Testing**: Confirm that changed classes were covered by unit tests.
- [x] **Integration Testing**: Confirm that changed logic was covered by integration tests.
- [x] **NEWS**: Confirm that the NEWS file is updated with relevant information about the changes made in this pull request.

### Related Issues
[MSEARCH-829](https://folio-org.atlassian.net/browse/MSEARCH-829)

### CODE REVIEW

File: src/main/java/org/folio/search/service/setter/linkeddata/authority/LinkedDataAuthorityLccnProcessor.java [hunk 1]
```diff
@@ -1,37 +1,24 @@
...
 import org.springframework.stereotype.Component;
 
 @Component
-@RequiredArgsConstructor
 public class LinkedDataAuthorityLccnProcessor implements FieldProcessor<LinkedDataAuthority, Set<String>> {
 
-  private final LccnNormalizer lccnNormalizer;
+  private final LinkedDataLccnProcessor linkedDataLccnProcessor;
+
+  public LinkedDataAuthorityLccnProcessor(@Autowired LccnNormalizer lccnNormalizer) {
```
Comment (AndreiBordak) [inline src/main/java/org/folio/search/service/setter/linkeddata/authority/LinkedDataAuthorityLccnProcessor.java:16]: @Autowired is not necessary in this case, is it?

**Review by AndreiBordak — APPROVED** [review]

File: src/main/java/org/folio/search/service/setter/linkeddata/authority/LinkedDataAuthorityLccnProcessor.java (hunk 1 above)
Comment (PBobylev) [reply src/main/java/org/folio/search/service/setter/linkeddata/authority/LinkedDataAuthorityLccnProcessor.java]: right, removed it, thank you!

File: src/main/java/org/folio/search/service/setter/linkeddata/instance/LinkedDataInstanceNoteProcessor.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  public Set<String> getFieldValue(LinkedDataInstance linkedDataInstance) {
+    var instanceNotes = ofNullable(linkedDataInstance.getNotes())
+      .stream()
+      .flatMap(Collection::stream)
+      .filter(Objects::nonNull);
+    var workNotes = ofNullable(linkedDataInstance.getParentWork())
+      .map(LinkedDataWorkOnly::getNotes)
+      .stream()
+      .flatMap(Collection::stream);
+    var contributors = Stream.concat(instanceNotes, workNotes).toList();
```
Comment (pkjacob) [inline src/main/java/org/folio/search/service/setter/linkeddata/instance/LinkedDataInstanceNoteProcessor.java:32]: `var notes`

**Review by pkjacob — APPROVED** [review]

File: src/main/resources/model/linked_data_instance.json [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      }
+    },
+    "parentWork": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "index": "keyword",
+          "searchAliases": [ "keyword" ]
+        },
+        "classifications": {
```
[the remark this answers is not in the document]
Comment (PBobylev) [reply src/main/resources/model/linked_data_instance.json:110]: But we show such information at the Search Result Page right now. I assumed that Instance search will be similar.
Let me check it with Doug

File: src/main/java/org/folio/search/service/setter/linkeddata/instance/LinkedDataInstanceNoteProcessor.java (hunk 2 above)
Comment (PBobylev) [reply src/main/java/org/folio/search/service/setter/linkeddata/instance/LinkedDataInstanceNoteProcessor.java:32]: right, fixed, thank you!

**Pull request merged**